src/sensors/icm20948sensor.cpp

```diff
@@ -359,174 +259,263 @@ void ICM20948Sensor::motionSetup() {
        m_Logger.fatal("Failed to reset FIFO");
         return;
     }
+}
 
-#if LOAD_BIAS
-    // Initialize the configuration
-    {
-        SlimeVR::Configuration::CalibrationConfig sensorCalibration = configuration.getCalibration(sensorId);
-        // If no compatible calibration data is found, the calibration data will just be zero-ed out
-        switch (sensorCalibration.type) {
-        case SlimeVR::Configuration::CalibrationConfigType::ICM20948:
-            m_Calibration = sensorCalibration.data.icm20948;
-            break;
-
-        case SlimeVR::Configuration::CalibrationConfigType::NONE:
-            m_Logger.warn("No calibration data found for sensor %d, ignoring...", sensorId);
-            m_Logger.info("Calibration is advised");
-            break;
-
-        default:
-            m_Logger.warn("Incompatible calibration data found for sensor %d, ignoring...", sensorId);
-            m_Logger.info("Calibration is advised");
-        }
+void ICM20948Sensor::connectSensor()
+{
+    #ifdef DEBUG_SENSOR
+        imu.enableDebugging(Serial);
+    #endif
+    // SparkFun_ICM-20948_ArduinoLibrary only supports 0x68 or 0x69 via boolean, if something else throw a error
+    boolean tracker = false;
+    
+    if (addr == 0x68) {
+        tracker = false;
+    } else if (addr == 0x69){
+        tracker = true;
+    } else {
+        m_Logger.fatal("I2C Address not supported by ICM20948 library: 0x%02x", addr);
+        return;
     }
-#endif
-
-    load_bias();
+    ICM_20948_Status_e imu_err = imu.begin(Wire, tracker);
+    if (imu_err != ICM_20948_Stat_Ok) {
+        m_Logger.fatal("Can't connect to ICM20948 at address 0x%02x, error code: 0x%02x", addr, imu_err);
+        ledManager.pattern(50, 50, 200);
+        return;
+    }
+}
 
+void ICM20948Sensor::startMotionLoop()
+{
     lastData = millis();
     working = true;
+}
 
-    #if defined(SAVE_BIAS) && SAVE_BIAS
-        timer.in(bias_save_periods[0] * 1000, [](void *arg) -> bool { ((ICM20948Sensor*)arg)->save_bias(true); return false; }, this);
-    #endif
+void ICM20948Sensor::checkSensorTimeout()
+{
+    if(lastData + 1000 < millis()) {
+        working = false;
+        lastData = millis();  
+        m_Logger.error("Sensor timeout I2C Address 0x%02x", addr);
+        Network::sendError(1, this->sensorId);
+    }
 }
 
-void ICM20948Sensor::motionLoop() {
-#if ENABLE_INSPECTION
+void ICM20948Sensor::checkForDataToRead(ICM_20948_Status_e readStatus)
```

Comment:
I'd prefer if this would just return a raw boolean instead of writing the result into a variable inside of the instance.